src/manager/status_manager.rs

```diff
@@ -105,6 +105,62 @@ pub async fn manage(
         // Get the time we started this so we can sleep for a specified time between queries
         let query_time = Instant::now();
         debug!("Starting status check");
+        // Update report statuses if reporting is enabled
+        if *config::ENABLE_REPORTING {
```

Comment:
I vaguely remember thinking about doing this at some point and deciding it would be difficult because reports, runs, and builds are all different types, and as far as I'm aware, there's no easy way to do something like, "If this variable is this type, call this function, but if it's this type, call this function, etc."